### PR TITLE
[X-1] Adding kafka transportation to auditing functionality

### PR DESCRIPTION
**Pull request by shipperizer**
@artsalliancemedia/thunderstorm

## Description

* Introducing kafka messaging in the auditing of requests
* using master version of ts messaging library just temporary until the functionality is `betable`
* Drop support for python3.5

### CODE REVIEW

**Comment by will-norris** [comment]
Add description pls

File: thunderstorm_auth/auditing.py [hunk 1]
```diff
@@ -15,11 +15,13 @@ class AuditConf(object):
...
+    def __init__(self, enabled=False, exclude_paths=None, kafka=False):
         """
         Args:
             enabled (bool or None): Defines whether or not auditing is enabled for API calls.
             exclude_paths (list): list of paths to exclude from auditing.
+            kafka (bool): Defines whether or not auditing is using kafka.
         """
         self.enabled = enabled or False
         self.exclude_paths = exclude_paths or []
+        self.kafka = kafka or False
```
Comment (will-norris) [inline thunderstorm_auth/auditing.py]: Could be just `self.kafka = kafka` - The value is a bool that defaults to False anyway. If they pass in something different like None thats their own mistake

File: thunderstorm_auth/falcon.py [hunk 2]
```diff
@@ -97,6 +97,7 @@ def process_response(self, req, res, resource, req_succeeded):
                 'status': res.status
             }
             try:
+                # TODO add kafka support, right now flacon apps don't use the authentication nor auditing so leaving it out
```
Comment (will-norris) [inline thunderstorm_auth/falcon.py]: *falcon

File: thunderstorm_auth/flask/core.py [hunk 3]
```diff
@@ -100,6 +114,36 @@ def after_request_auditing(response):
...
         app.extensions['ts_auth'] = self.state
 
+    def init_kafka(self):
+        """
+        Initialize kafka with app.config variables
+
+        Returns:
+                TSKafka instance
+        """
+        # incoming format is 'kafka-1,kafka-2,kafka-3'
```
Comment (will-norris) [inline thunderstorm_auth/flask/core.py]: incoming format is 'kafka-1:9092,kafka-2:9092,kafka-3:9092'

File: requirements.txt [hunk 4]
```diff
@@ -6,5 +6,5 @@ requests>=2.20.0<3
 sqlalchemy>=1.1,<2
 marshmallow>=2.15,<4
 werkzeug<1
-thunderstorm_library @ git+https://github.com/artsalliancemedia/thunderstorm-library@v1.4.0#egg=thunderstorm_library-1.4.0
+thunderstorm_library @ git+https://github.com/artsalliancemedia/thunderstorm-library@master#egg=thunderstorm_library-master
```
Comment (will-norris) [inline requirements.txt]: Maybe put a # TODO comment above this line as a reminder to switch back to the versioned library when it is ready. Easy for changes like this to get lost when other stuff is going on

File: thunderstorm_auth/falcon.py (hunk 2 above)
Comment (shipperizer) [reply thunderstorm_auth/falcon.py]: ###FLACON

**Pull request merged**